### ISSUE TITLE
Completely rework pointer management.

### ISSUE THREAD

**Issue by lobingera** [JuliaGraphics/Cairo.jl#320]
Following the discussion in https://discourse.julialang.org/t/avoid-gc-freeing-ptr-nothing/38664/15
(yuyichao) recommends major changes.

(me wonders, why this went undetected for so long ...)

**Comment by donm** [comment]
I believe this is related to #252?

**Comment by lobingera** [comment]
No. It's (roughly) about how GC tracks if memory is still in use and Cairo.jl today uses some ideas about pointer management that (according to yuyichao) should have never worked (which they did surprisingly well for long time). I started and stopped working on this locally.

**Comment by donm** [comment]
You're definitely right.  I can't even retrace my thoughts that made me think they were related, to the point that I think I copy/pasted the wrong issue number, or entered that message in the wrong issue.

Since I've already bothered you for no good reason, thanks for your work on Cairo.jl.
